frontends/common/options.cpp

```diff
@@ -130,6 +133,38 @@ void CompilerOptions::setInputFile() {
     }
 }
 
+std::vector<const char*>* CompilerOptions::process(int argc, char* const argv[]) {
+    char buffer[PATH_MAX];
+    int len;
+    struct stat st;
+    if ((len = readlink("/proc/self/exe", buffer, sizeof(buffer))) > 0 ||
```

Comment:
Its portable to Linux, *BSD, and Solaris, and it won't fail badly on other platforms.  The fallback using argv[0] will work on OSX and I believe on Windows (cygwin/msys), though I haven't tried the latter